packages/redux-genes/src/GenePageHoc.js

```diff
@@ -1,85 +1,148 @@
-/* eslint-disable react/no-unused-prop-types */
-/* eslint-disable no-shadow */
-/* eslint-disable import/no-unresolved */
-/* eslint-disable import/extensions */
-
-import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import React, { Component } from 'react'
 import { connect } from 'react-redux'
 
 import { actions as variantActions } from '@broad/redux-variants'
-import { actions as tableActions } from '@broad/table'
-import { currentGene, currentTranscript, isFetching, geneData, geneNotFound, actions as geneActions } from './index'
+import { Loading } from '@broad/ui'
+
+import { actions as geneActions } from './genes'
+
 
 const GenePageContainer = ComposedComponent => class GenePage extends Component {
   static propTypes = {
-    currentGene: PropTypes.string,
-    gene: PropTypes.object,
-    isFetching: PropTypes.bool.isRequired,
-    fetchGeneIfNeeded: PropTypes.func.isRequired,
-    resetSearchVariants: PropTypes.func.isRequired,
+    fetchGeneData: PropTypes.func.isRequired,
+    geneName: PropTypes.string.isRequired,
+    transcriptId: PropTypes.string,
   }
 
   static defaultProps = {
-    gene: null,
-    currentGene: null,
+    transcriptId: undefined,
+  }
+
+  state = {
+    geneData: null,
+    isLoading: false,
+    loadError: null,
   }
 
   componentDidMount() {
-    const { currentGene, match, fetchGeneIfNeeded } = this.props
-    fetchGeneIfNeeded(currentGene, match, history)
+    this.mounted = true
+    this.loadGeneData()
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { fetchGeneIfNeeded, currentGene, currentTranscript, history } = this.props
-    if (currentGene !== nextProps.currentGene ||
-      currentTranscript !== nextProps.currentTranscript
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.geneName !== prevProps.geneName
+      || this.props.transcriptId !== prevProps.transcriptId
     ) {
-      // if(this.props.route.path == nextProps.route.path) return false
-      history.push(`/gene/${nextProps.currentGene}`)
-      fetchGeneIfNeeded(nextProps.currentGene)
-      this.props.resetSearchVariants()
-      this.props.resetFilter()
+      this.loadGeneData()
     }
   }
 
+  componentWillUnmount() {
+    this.mounted = false
+  }
+
+  loadGeneData() {
+    this.setState({
+      isLoading: true,
+      loadError: null,
+    })
+
+    this.props.fetchGeneData(this.props.geneName, this.props.transcriptId)
+      .then((geneData) => {
+        let loadError = null
+        if (!geneData) {
+          loadError = 'Gene not found'
+        }
+        if (
+          this.props.transcriptId
+          && !geneData.transcripts.map(t => t.transcript_id).includes(this.props.transcriptId)
```

Comment:
This is a quick fix for handling "transcript not found" but isn't great since it relies on `fetchGeneData` requesting transcripts for the gene.